core/iwasm/aot/aot_runtime.c

```diff
@@ -1310,11 +1310,53 @@ invoke_native_with_hw_bound_check(WASMExecEnv *exec_env, void *func_ptr,
     (void)jmpbuf_node_pop;
     return ret;
 }
+#endif /* end of OS_ENABLE_HW_BOUND_CHECK */
+
+#ifndef OS_ENABLE_BLOCK_INSN_INTERRUPT
 
+#ifdef OS_ENABLE_HW_BOUND_CHECK
 #define invoke_native_internal invoke_native_with_hw_bound_check
-#else /* else of OS_ENABLE_HW_BOUND_CHECK */
+#else
 #define invoke_native_internal wasm_runtime_invoke_native
-#endif /* end of OS_ENABLE_HW_BOUND_CHECK */
+#endif
+
+#else /* else of OS_ENABLE_BLOCK_INSN_INTERRUPT */
+
+#ifdef OS_ENABLE_HW_BOUND_CHECK
+#define invoke_native_block_insn_interrupt invoke_native_with_hw_bound_check
+#else
+#define invoke_native_block_insn_interrupt wasm_runtime_invoke_native
+#endif
+
+static bool
+invoke_native_internal(WASMExecEnv *exec_env, void *func_ptr,
+                       const WASMType *func_type, const char *signature,
+                       void *attachment, uint32 *argv, uint32 argc,
+                       uint32 *argv_ret)
+{
+    int ret = false;
+    WASMJmpBuf jmpbuf_node = { 0 }, *jmpbuf_node_pop;
+    wasm_exec_env_push_jmpbuf(exec_env, &jmpbuf_node);
+    wasm_runtime_set_exec_env_tls(exec_env);
+
+    if (os_setjmp(jmpbuf_node.jmpbuf) == 0) {
+        ret = invoke_native_block_insn_interrupt(exec_env, func_ptr, func_type,
+                                                 signature, attachment, argv,
+                                                 argc, argv_ret);
+    }
+    else {
+        ret = false;
```

Comment:
this line wouldn't be needed, right? as ret is false by default

Reply:
https://github.com/bytecodealliance/wasm-micro-runtime/pull/1930#discussion_r1094289179